.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,289 @@
+    with:
+      directory: hack/helm-reference-gen
+      go-version: 1.17.2
+
+  unit-test-helm-templates:
+    needs: [unit-helm-gen]
+    runs-on: ubuntu-latest
+    container: 
+      image: docker.mirror.hashicorp.services/hashicorpdev/consul-helm-test:0.10.0
+      options: --user root
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+   
+      - name: Run Unit Tests
+        working-directory: charts/consul
+        run:  bats --jobs 4 ./test/unit
+
+  lint-control-plane:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.2
+
+      - run: go get -u github.com/hashicorp/lint-consul-retry && lint-consul-retry
+
+      - name: Run lint
+        working-directory: control-plane
+        run: go run hack/lint-api-new-client/main.go
+
+  go-fmt-vet-control-plane:
+    uses: hashicorp/consul-k8s/.github/workflows/reusable-go-fmt-vet.yml@crt-build-and-tests
+    with:
+      directory: control-plane
+      go-version: 1.17.2
+
+  test-control-plane:
+    needs: [lint-control-plane, go-fmt-vet-control-plane]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.2
+
+      - name: Setup go mod cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install gotestsum
+        run: |
+          wget https://github.com/gotestyourself/gotestsum/releases/download/v${{env.GOTESTSUM_VERSION}}/gotestsum_${{env.GOTESTSUM_VERSION}}_linux_amd64.tar.gz
+          sudo tar -C /usr/local/bin -xzf gotestsum_${{env.GOTESTSUM_VERSION}}_linux_amd64.tar.gz
+          rm gotestsum_${{env.GOTESTSUM_VERSION}}_linux_amd64.tar.gz
+
+      - run:  mkdir -p ${{env.TEST_RESULTS}}
+      - run:  echo "$HOME/bin" >> $GITHUB_PATH
+
+      - name: Run go tests 
+        working-directory: control-plane 
+        run: |
+            mkdir -p $HOME/bin
+            wget https://releases.hashicorp.com/consul/${{env.CONSUL_VERSION}}/consul_${{env.CONSUL_VERSION}}_linux_amd64.zip && \
+              unzip consul_${{env.CONSUL_VERSION}}_linux_amd64.zip -d $HOME/bin && \
+              rm consul_${{env.CONSUL_VERSION}}_linux_amd64.zip
+            chmod +x $HOME/bin/consul
+            PACKAGE_NAMES=$(go list ./...)
+            gotestsum --junitfile ${{env.TEST_RESULTS}}/gotestsum-report.xml -- -p 4 $PACKAGE_NAMES
+
+  test-enterprise-control-plane:
+    needs: [lint-control-plane, go-fmt-vet-control-plane]
+    runs-on: ubuntu-latest
+    env:
+      CONSUL_LICENSE: ${{secrets.CONSUL_LICENSE}}
```

Comment:
You have to specifically pull in the secrets you want into the workflow, unlike CircleCI where it is just available to you as an environment variable.